### PR TITLE
feat: tags and tabbuttongroup components

### PR DESCRIPTION
**Pull request by GitAndrew**
Example use:
const [selected, setSelected] = useState(0);
....
     <TabButtonGroup
        buttons={["one", "two", "three"]}
        selectedIndex={selected}
        onPress={(selectedIndex: number) => {
          setSelected(selectedIndex);
        }}
      />
      <Tags
        tags={[{ title: "apple" }, { title: "banana" }, { title: "orange" }]}
        variant="square"
        color="orange"
      />

### CODE REVIEW

File: src/components/ui/TabButtonGroup/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+interface TabButtonGroupProps {
+  id?: string;
+  onPress: (index: number) => void;
+  selectedIndex: number;
+  buttons: string[];
+}
+
+const TabButtonGroup: React.FC<TabButtonGroupProps> = ({
+  id,
```
Comment (winniezheng) [inline src/components/ui/TabButtonGroup/index.tsx:16]: don't hate us but we started destructuring the props...

File: src/components/ui/TabButtonGroup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            key={`${index}-${button}`}
+            data-testid={`${index}-${button}-btn`}
+            title={button}
+            onPress={() => onPress(index)}
+            type={selected ? "solid" : "outline"}
+            containerStyle={{
+              marginRight: RFValue(10),
+              
+            }}
+            buttonStyle={{ backgroundColor: selected ? colors.darkBrown : colors.white, borderColor: colors.darkBrown }}
```
Comment (winniezheng) [inline src/components/ui/TabButtonGroup/index.tsx:36]: how come you aren't defining styles in styles.ts

Comment (chrisrabe) [reply src/components/ui/TabButtonGroup/index.tsx:36]: According to these sources, you can pass props into the styles variable by making it into a callback function

https://stackoverflow.com/questions/42707327/passing-props-into-external-stylesheet-in-react-native
https://cloudstack.ninja/arahpanah/how-to-pass-props-to-stylesheet-in-react-native-using-typescript/

File: src/components/ui/Tags/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const Tags: React.FC<TagsProps> = ({ tags, id, variant = "round", color = "orange" }) => {
+  return (
+    <View data-testid={id} style={styles.container}>
+      {tags.map(({ id, title, ...rest }, index) => (
+        <Chip
+          key={`${index}-${title}`}
+          data-testid={id}
+          title={title}
+          {...rest}
+          buttonStyle={{ borderRadius: variant === "round" ? 30 : 3, marginRight: RFValue(10), backgroundColor: colors[color] }}
```
Comment (chrisrabe) [inline src/components/ui/Tags/index.tsx:29]: Same comment as above

File: src/components/ui/TabButtonGroup/index.tsx (hunk 2 above)
Comment (GitAndrew) [reply src/components/ui/TabButtonGroup/index.tsx:36]: buttonStyle is a prop from Button. I can try just passing style, but not sure if it'll work

File: src/components/ui/TabButtonGroup/index.tsx (hunk 1 above)
Comment (GitAndrew) [reply src/components/ui/TabButtonGroup/index.tsx:16]: ok

Comment (winniezheng) [reply src/components/ui/TabButtonGroup/index.tsx:16]: is it ok

Comment (winniezheng) [reply src/components/ui/TabButtonGroup/index.tsx:16]: maybe we should have a meeting about conventions

File: src/components/ui/Tags/index.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+interface TagsProps {
+  tags: Tag[];
+  id?: string;
+  variant?: "round" | "square";
+  color?: "teal" | "lightbrown" | "orange";
+}
+
+const Tags: React.FC<TagsProps> = ({ tags, id, variant = "round", color = "orange" }) => {
```
Comment (chrisrabe) [inline src/components/ui/Tags/index.tsx:20]: Can we use defaultProps for prop types with default values?

https://blog.bitsrc.io/understanding-react-default-props-5c50401ed37d

Comment (GitAndrew) [reply src/components/ui/Tags/index.tsx:20]: For functional components, I'd say defaultProps is outdated and may potentially? get deprecated in the future
https://stackoverflow.com/questions/41488537/react-defaultprops-vs-es6-default-params-when-destructuring-performances-issu

**Review by chrisrabe — APPROVED** [review]

**Pull request merged**